sentry_sdk/scope.py

```diff
@@ -249,6 +252,26 @@ def clear_breadcrumbs(self):
         """Clears breadcrumb buffer."""
         self._breadcrumbs = deque()  # type: Deque[Breadcrumb]
 
+    def add_attachment(
+        self,
+        bytes=None,  # type: Optional[bytes]
+        filename=None,  # type: Optional[str]
+        path=None,  # type: Optional[str]
+        content_type=None,  # type: Optional[str]
+        add_to_transactions=False,  # type: bool
+    ):
+        # type: (...) -> None
+        """Adds an attachment to future events sent."""
+        self._attachments.append(
+            Attachment(
+                bytes=bytes,
+                path=path,
+                filename=filename,
+                content_type=content_type,
+                add_to_transactions=add_to_transactions,
+            )
+        )
```

Comment:
We discussed off GH the idea of having different constructors to disambiguate usage. We either do that or should have at least some documentation around how these arguments are supposed to be used or not used together.

Alternative proposal, if we remove `filename`, we may be able to remove the ambiguity and the need for multiple constructors:

```suggestion
    def attach(
        self,
        path,  # type: str
        bytes=None,  # type: Optional[bytes]
        content_type=None,  # type: Optional[str]
        add_to_transactions=False,  # type: bool
    ):
        # type: (...) -> None
        """Adds an attachment to future events sent.
        
        The path must point to a file in disk which must be readable at the time
        of sending events, unless bytes is provided. When passing explicit bytes,
        path will not be read and therefore does not need to exist nor be readable.
        
        If path points to a file that is expected to mutate over time or may become
        inaccessible, it is recommended to read it and pass in explicit bytes.
        """
        self._attachments.append(
            Attachment(
                path,
                bytes=bytes,
                content_type=content_type,
                add_to_transactions=add_to_transactions,
            )
        )
```

Reply:
Resolving as duplicate of the above conversation.